File: Neural/Node.py
from Constants import PieceType


class Node:

    def __init__(self, board=None, piece=None):
        self.value = -1
        self.children = []
        self.board = [[]]
        self.updated_piece = PieceType.Empty

        if board is not None:
            self.board = board
        if piece is not None:
            self.updated_piece = piece

File: MiniMax/MiniMax_TicTacToe_Lib/Node.cs
using System.Collections.Generic;

namespace MiniMax_TicTacToe_Lib
{
    public class Node
    {
        public readonly PieceType UpdatedPiece;

        public int Value;

        public readonly PieceType[][] Board;
        public List<Node> Children;

        public Node()
        {
            Value = -1;
            Children = new List<Node>();
        }

        public Node(PieceType[][] _board, PieceType _piece) : this()
        {
            Board = _board;
            UpdatedPiece = _piece;
        }
    }
}